### ISSUE TITLE
JSON Parser drops decimal precision

### ISSUE THREAD

**Issue by joynerar** [jOOQ/jOOQ#12173]
### Expected behavior

When converting a JSON string into a JSON object, the resulting JSONB object will drop any trailing zeroes for a decimal value.

So if the following string was passed in: 
`JSONB.valueOf("{"length" : 6.67000}")`

the resulting object should retain the trailing zeroes:
```
{
    "length" : 6.67000
}
```

### Actual behavior

 Passing in: 
`JSONB.valueOf("{"length" : 6.67000}")`

Will create the an object as such: 
```
{
    "length" : 6.67
}
```

This is necessary in mathematical / scientific applications where precision is expected to be maintained. 

- If the problem relates to code generation, please post your code generation configuration
- If the problem relates to upgrades, please check if your RDBMS version is still supported by jOOQ: https://www.jooq.org/download/support-matrix
- A complete set of DDL statements can help re-create the setup you're having
- An MCVE can be helpful to provide a complete reproduction case: https://github.com/jOOQ/jOOQ-mcve

### Versions

- jOOQ: 3.15.0
- Java: 11
- Database (include vendor): Postgres 10.14
- OS: Windows 10 
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your report. That's probably the same issue as https://github.com/jOOQ/jOOQ/issues/12144, which has been fixed for 3.16.0 and 3.15.1 snapshot versions already. Can you confirm this? If you're licensed, there are nightly builds available from here:
https://www.jooq.org/download/versions

Otherwise, you can build from github, or apply these patches to check:

- https://github.com/jOOQ/jOOQ/commit/e974a5e07273076feeb691521a8afc064135c7f7
- https://github.com/jOOQ/jOOQ/commit/46a49e24aad4df6a97e92548727073ede20edc63

3.15.1 should be released today, so it would be great if you can confirm your issue is already fixed by #12144.

Note, to my understanding, a JSON number works like a JavaScript number, i.e. it is a floating point number value, not a decimal number value. E.g. our parser library that we're using (and which has caused this issue here) uses `java.lang.Double` to represent numbers: https://github.com/jOOQ/jOOQ/issues/12176

If you want to stay on the safe side (across libraries, languages, etc.), perhaps it's better to represent your number as a JSON string?

**Comment by lukaseder** [comment]
Note that even with the fix #12176, there will be a difference between calling `JSONB.toString()` and `JSONB.data()`:

```java
System.out.println(ctx.select(val(JSONB.valueOf("{\"length\": 6.67000}"))).fetchOne().value1().toString());
System.out.println(ctx.select(val(JSONB.valueOf("{\"length\": 6.67000}"))).fetchOne().value1().data());
```

Prints:

```
{"length":6.67}
{"length": 6.67000}
```

All of this change of behaviour is caused by #12023

**Comment by lukaseder** [comment]
See also PostgreSQL:

```sql
select '{"length":6.67}'::jsonb = '{"length": 6.67000}'::jsonb
```

Which produces:

``` 
true
```
